CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.22)
+project(NEBULA)
+
+set(CMAKE_CXX_STANDARD 11)
+
+set(CMAKE_CXX_COMPILER "g++")
```

Comment:
what if I use clang?

Reply:
CMAKE_CXX_COMPILER may be set at setup time via cli defines. no need for custom options